src/cgns_internals.c

```diff
@@ -4530,26 +5463,15 @@ int cgi_read_exponents(int in_link, double parent_id, cgns_exponent **exponents)
     return CG_OK;
 }
 
-int cgi_read_units(int in_link, double parent_id, cgns_units **units)
+int cgi_read_units_node(int in_link, cgns_units** units)
 {
     char_33 unit_name;
-    char *string_data;
-    double *id;
+    char* string_data;
+    double* id;
     int nnod;
 
-    if (cgi_get_nodes(parent_id, "DimensionalUnits_t", &nnod, &id)) return CG_ERROR;
-    if (nnod<=0) {
-        units[0]=0;
-        return CG_OK;
-    }
-    units[0] = CGNS_NEW(cgns_units, 1);
-    units[0]->id = id[0];
-    units[0]->link = cgi_read_link(id[0]);
-    units[0]->in_link = in_link;
-    CGNS_FREE(id);
-
     if (cgi_read_string(units[0]->id, units[0]->name, &string_data)) return CG_ERROR;
-    if (strlen(string_data) != 32*5) {
+    if (strlen(string_data) != 32 * 5) {
```

Comment:
Maybe replace the `32` magic number with some meaningful name...  Maybe that is a separate PR later on to not clutter this one...

Reply:
I haven't investigated from where comes this magic number. You are right a separate PR would be needed.